pandas/core/frame.py

```diff
@@ -2297,10 +2297,10 @@ def to_markdown(
         result = tabulate.tabulate(self, **kwargs)
         if buf is None:
             return result
-        ioargs = get_filepath_or_buffer(buf, mode=mode, storage_options=storage_options)
-        assert not isinstance(ioargs.filepath_or_buffer, (str, mmap.mmap))
-        ioargs.filepath_or_buffer.writelines(result)
-        ioargs.close()
+        handles = get_handle(buf, mode, storage_options=storage_options)
```

Comment:
it would be great if we could call `get_handle` as a context manager (not sure if that would bloat the size of this PR) or as a followup ok

Reply:
I think this is possible and would also be a fairly small change (add enter and exit function to IOHandles).

Could then be used 
```py
with get_handles(...) as handles:
    handles.handel.write(...)
```

I will think about that and either add it here or more likely have a followup.